plugins/de.cau.cs.kieler.klighd/src/de/cau/cs/kieler/klighd/internal/macrolayout/KlighdDiagramLayoutConnector.java

```diff
@@ -377,9 +378,26 @@ private void createNode(final LayoutMapping mapping, final KNode node,
 
         // KLighD is somewhat mean and doesn't care about existing insets
         node.setInsets(insets);
-        node.getProperty(CoreOptions.PADDING);
         // The Insets are used in {@link KlighdLayoutConfigurationStore} to retrieve the padding
         // of the node
+        
+        if (node.getProperty(KlighdProperties.ROUNDED_RECTANGLE_AUTOPADDING) != null) {
+            KVector radii = node.getProperty(KlighdProperties.ROUNDED_RECTANGLE_AUTOPADDING);
+            double padding = 0;
+            // x and y cannot be zero, if one of the two is zero use the larger value
+            if (radii.x < 0.00001 && radii.y >= 0.00001) {
+                radii.x = 0.00001;
+            } else if (radii.x >= 0.00001 && radii.y < 0.00001) {
+                radii.y = 0.00001;
+            }
+            if (!(radii.x < 0.00001 && radii.y < 0.00001)) {
```

Comment:
please put these checks either in a `almostZero(...)` method or extract this as a `almostZero` constant.